### ISSUE TITLE
Remove wl_list of resources in many protocols

### ISSUE THREAD

**Issue by ascent12** [swaywm/wlroots#1539]
A bunch of our interfaces have lists of resources which don't need to be there.
For example
```c
struct wlr_compositor {
    struct wl_global;
    struct wl_list resources; // this
    struct wl_list surface_resources; // this
    struct wl_list subsurface_resources; // this
    // ...
}
```
I'm not aware of anybody using these resource lists for anything or why they would need to be used.
We don't need to keep track of these for clean-up, because calling `wl_display_destroy_clients` will go through and destroy the resources through the normal mechanism.

**Comment by emersion** [comment]
+1 (though there are some cases in which we need to keep track of them)

Related to https://github.com/swaywm/wlroots/issues/1411

**Comment by ascent12** [comment]
Implemented in #1915
